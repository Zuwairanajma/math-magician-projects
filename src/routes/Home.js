import LoremIpsum from 'react-lorem-ipsum';
import '../styles/Home.css';

function Home() {
  return (
    <div>
      <h2 style={{ margin: '60px 140px' }}>Welcome to our page!</h2>
      <div className="text-wrapper">
        <LoremIpsum avgSentencesPerParagraph={10} random={false} p={2} />
      </div>
    </div>
  );
}

export default Home;
