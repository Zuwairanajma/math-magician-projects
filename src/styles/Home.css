.text-wrapper {
  width: 80%;
  margin: 10px auto;
}

p {
  margin-bottom: 20px;
}
