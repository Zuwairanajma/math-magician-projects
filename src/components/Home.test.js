import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/jest-globals';
import Home from './Home';

describe('Testing Home', () => {
  test('render with snapshot', () => {
    const element = render(<Home />);
    expect(element).toMatchSnapshot();
  });

  test('Render Home component', () => {
    const { getByTestId } = render(<Home />);
    const element = getByTestId('display');
    expect(element).toBeInTheDocument();
  });
});
