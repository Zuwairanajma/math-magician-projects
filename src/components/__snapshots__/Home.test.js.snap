// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home render with snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="homepage"
        data-testid="display"
      >
        <h1
          class="home-title"
        >
          Welcome to our page!
        </h1>
        <p
          class="home-paragrhap"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatem, doloremque perferendis. Eos porro harum libero voluptatem magnam? Alias voluptatem porro quas ullam. Voluptates vel ad incidunt illo ducimus explicabo beatae! Ex neque distinctio, rem earum nam optio molestias. Facilis odit quidem est recusandae numquam nemo totam alias voluptatem, minima minus, eaque, repellat perspiciatis! Tenetur totam veniam reprehenderit, dignissimos rem quia!
        </p>
        <p
          class="home-paragrhap"
        >
          Debitis, cumque nobis possimus reprehenderit dolorem nostrum aut odio libero ipsa, consequuntur nisi vitae incidunt voluptates repellat dolor. Corrupti obcaecati eveniet est pariatur eos esse animi ut explicabo dicta earum. Mollitia enim dolore delectus, ipsum, esse tempora, sunt commodi consectetur nobis cum at rerum praesentium asperiores nihil voluptate saepe est accusamus maxime ducimus! Nemo debitis fugiat provident, reprehenderit quaerat ratione.
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="homepage"
      data-testid="display"
    >
      <h1
        class="home-title"
      >
        Welcome to our page!
      </h1>
      <p
        class="home-paragrhap"
      >
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatem, doloremque perferendis. Eos porro harum libero voluptatem magnam? Alias voluptatem porro quas ullam. Voluptates vel ad incidunt illo ducimus explicabo beatae! Ex neque distinctio, rem earum nam optio molestias. Facilis odit quidem est recusandae numquam nemo totam alias voluptatem, minima minus, eaque, repellat perspiciatis! Tenetur totam veniam reprehenderit, dignissimos rem quia!
      </p>
      <p
        class="home-paragrhap"
      >
        Debitis, cumque nobis possimus reprehenderit dolorem nostrum aut odio libero ipsa, consequuntur nisi vitae incidunt voluptates repellat dolor. Corrupti obcaecati eveniet est pariatur eos esse animi ut explicabo dicta earum. Mollitia enim dolore delectus, ipsum, esse tempora, sunt commodi consectetur nobis cum at rerum praesentium asperiores nihil voluptate saepe est accusamus maxime ducimus! Nemo debitis fugiat provident, reprehenderit quaerat ratione.
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
